### PR TITLE
Add Client ssl validation

### PR DESCRIPTION
**Pull request by bleuchtang**
Add to nginx the availability to check client certificate.

### CODE REVIEW

File: templates/etc/nginx/sites-available/default.conf.j2 [hunk 1]
```diff
@@ -980,6 +995,15 @@ server {
         ssl_stapling_verify       on;
         ssl_trusted_certificate   {{ nginx_tpl_ssl_trusted_certificate }};
 {% endif %}
+{% if (item.ssl_verify_client | d(nginx_default_ssl_verify_client)) | bool %}
+{% if ((item.ssl_client_certificate is defined and item.ssl_client_certificate) or nginx_default_ssl_client_certificate) %}
+        ssl_verify_client         on;
+        ssl_client_certificate    {{ item.ssl_client_certificate | default(nginx_default_ssl_client_certificate) }};
+{% endif %}
+{% if ((item.ssl_crl is defined and item.ssl_crt) or nginx_default_ssl_crl) %}
```
Comment (drybjed) [inline templates/etc/nginx/sites-available/default.conf.j2]: This should probably be
```
{% if (item.ssl_crl|d() or nginx_default_ssl_crl) %}

File: templates/etc/nginx/sites-available/default.conf.j2 [hunk 2]
```diff
@@ -177,6 +177,21 @@
...
+#
+#
+#   - item.ssl_client_certificate: ''
+#      Optional. Specifies a file with trusted CA certificates in the PEM
+#      format used to verify client certificates.
+#
+#
+#   - item.ssl_crl: ''
+#      Optional. Specifies a file with revoked certificates (CRL) in the PEM
+#      format used to verify client certificates.
```
Comment (ypid) [inline templates/etc/nginx/sites-available/default.conf.j2]: This documentation format is deprecated [as it says on the top](https://github.com/debops/ansible-nginx/blob/master/templates/etc/nginx/sites-available/default.conf.j2#L8-L9). Can you move this to [defaults-detailed.rst](https://github.com/debops/ansible-nginx/blob/master/docs/defaults-detailed.rst#nginx-servers)?

Comment (drybjed) [inline templates/etc/nginx/sites-available/default.conf.j2]: Sure, that's the plan when I get around to `debops.nginx` again. :-)

**Comment by ypid** [comment]
@bleuchtang Nice work :+1:

**Comment by drybjed** [comment]
Looks good, although I would like to see a better integration with `debops.pki` realms. That can be done later though. I'll merge this PR shortly.

**Pull request merged**